Upgrade rails to 6.1.4

Yuri-SN opened the pull request.

santib reviewed (commented):
Hi @JuraSN, thanks for the contribution.

I think as part of the upgrade we also need to run `bin/rails app:update` to update our config files. For example in `config/puma.rb` there is a section that was removed, in `config/environments/production.rb` there are some new options even though they are commented out by default, etc.

If you don't have enough context to do this, maybe we can merge your branch to a feature branch on our repo so I can add another commit doing what I said before. WDYT?

JuraSN commented:
Hello, @santib!

I ran the command: ```bin/rails app:update``` and force rewrite all files in project.

And then:

* added the necessary changes from the project files
* updated the Rubocop gem and fixed all the warnings and errors in the code analysis
* ran rspec tests and made sure that everything is fine

santib commented on config/environments/production.rb:
```diff
@@ -23,19 +25,18 @@
-  # config.assets.css_compressor = :sass
-
-  # Fallback to assets pipeline if a precompiled asset is missed.
-  config.assets.compile = true
```
not sure we should remove this. In Rails code I see

```erb
  <%- unless options.skip_sprockets? -%>
  # Compress CSS using a preprocessor.
  # config.assets.css_compressor = :sass

  # Do not fallback to assets pipeline if a precompiled asset is missed.
  config.assets.compile = false

  <%- end -%>
```

and I think we are using `sprockets` for the admin site.

Yuri-SN replied:
Returned back this option.

santib commented on config/initializers/new_framework_defaults_6_1.rb:
```diff
@@ -0,0 +1,67 @@
+# Be sure to restart your server when you modify this file.
```
This file should be removed since we are already loading 6.1 defaults in `config/application.rb`. The purpose of this file is helping migrating to the new defaults

Yuri-SN replied:
Ok.

santib reviewed (approved):
Beautiful work 👏  Left a couple last comments

Thanks a lot for working on this

federicoaldunate commented on Gemfile.lock:
```diff
@@ -424,37 +401,30 @@ GEM
       crack (>= 0.3.2)
       hashdiff (>= 0.4.0, < 2.0.0)
-    webpacker (5.1.1)
+    webpacker (5.4.2)
```
Given that Webpacker was upgraded, we need to upgrade them in `yarn.lock` and `package.json`

Yuri-SN replied:
Done.

Yuri-SN commented:
* connected commits to one
* in **config/environments/production.rb** returned back **config.assets.compile = true** and **config.assets.digest = true**
* deleted config/initializers/new_framework_defaults_6_1.rb
* upgraded yarn.lock and package.json
* added **config/master.key** to **.gitignore**

santib reviewed (approved):
Amazing job! Thanks @Yuri-SN 🙌

martinjaimem reviewed (approved):
Nice improvement @Yuri-SN !

Thanks so much for contributing. 
I looked at it with @santib and it looks perfect.

There's one more thing we should do in order to merge the pr, you should run `i18n-tasks normalize` as it changes a translation file. You can take a look at it here: https://github.com/rootstrap/rails_api_base/pull/324.

Feel free to make the change yourself and we merge your PR! we just did a separate PR because the CI got stuck and wasn't running in your branch.

Yuri-SN commented:
Hello, @martinjaimem!

> There's one more thing we should do in order to merge the pr, you should run `i18n-tasks normalize` as it changes a translation file. You can take a look at it here: #324.

I looked at the changes in #324, run `i18n-tasks normalize` and commit changes with --amend flag.
